### PR TITLE
chore(style): improved prettier config and add stylelint

### PR DESCRIPTION
**Pull request by ghost**
## PR Area

- [x] Frontend
- [ ] Testing
- [ ] Other: 

## Type of change

_What kind of change does this PR introduce?_

- [ ] Bugfix
- [ ] Feature
- [x] Other: formatting

## Description
* improve prettier config to be more consistent with internal standards
* adds stylelint
* format components to adhere to rules
* fix warnings

Note: run `yarn`, then in VSCode `Developer: Reload Window` after pulling this to ensure rules are applied

## Security & risks

- [x] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/components/dialog/Dialog.vue [hunk 1]
```diff
@@ -43,6 +43,7 @@ export default defineComponent({
       default: 'Close',
     },
   },
+  emits: ['toggle-dialog'],
```
Comment (ghost) [inline src/components/dialog/Dialog.vue:46]: @arjun-at-roam Note this change. It's required to add an `emits` array if having a method that calls `emit('some-custom-event')` – we'll now get proper warnings for this

File: .stylelintrc.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "no-descending-specificity": null,
+    "no-duplicate-selectors": null,
+    "at-rule-empty-line-before": "always",
+    "rule-empty-line-before": [
+      "always-multi-line",
+      { "except": ["first-nested", "after-single-line-comment"] }
+    ],
+    "selector-pseudo-element-no-unknown": [true, {
+      "ignorePseudoElements": ["v-slotted"]
+    }]
```
Comment (ghost) [inline .stylelintrc.json:19]: Important config for using stylelint with Vue. There may be other Vue-specific pseudo elements that will need to be added here in the future

File: src/providers/Theme.vue [hunk 3]
```diff
@@ -25,6 +25,10 @@ export default defineComponent({
 
   // Shared properties
   --font-base: 'proxima nova', sans-serif;
+  --font-secondary: -apple-system, BlinkMacSystemFont, Helvetica, Arial,
```
Comment (arjun-at-roam) [inline src/providers/Theme.vue]: should we stick to web safe?

Comment (ghost) [reply src/providers/Theme.vue]: This is web safe when considering the fallbacks. It's the font used in the designs. It's the macOS system font if using macOS. But I will check with Danielle if we should change this font. Let's leave it as is for now, it's not a blocker for this PR.

**Review by arjun-at-roam — APPROVED** [review]

**Review by antoninroam — APPROVED** [review]

**Review by jpyepez-roam — APPROVED** [review]
Nice setup! LGTM! 🚀

**Pull request merged**